Okta/OktaAuth/OktaAppAuth.swift

```diff
@@ -10,67 +10,71 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import Hydra
-
 // Current version of the SDK
 let VERSION = "2.1.0"
 
 // Holds the browser session
 public var currentAuthorizationFlow: OIDExternalUserAgentSession?
 
 // Cache Okta.plist for reference
-public var configuration: [String: Any]?
+public var configuration = try? OktaAuthConfig.default()
 
 // Cache the Discovery Metadata
 public var wellKnown: [String: Any]?
 
+internal var authApi: OktaApi = OktaApiImpl()
+
 // Token manager
-public var tokens = OktaTokenManager.readFromSecureStorage()
+public var tokens = OktaTokenManager.readFromSecureStorage() {
+    didSet {
+        tokens?.writeToSecureStorage()
+    }
+}
 
-public func signInWithBrowser() -> SignIn {
-    // Authenticate via authorization code flow
-    return SignIn()
+public var isAuthenticated: Bool {
+    return tokens?.accessToken != nil
 }
 
-public func signOutOfOkta() -> SignOut {
-    // End the Okta session
-    return SignOut()
+public func signInWithBrowser(from presenter: UIViewController, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
+    SignInTask(config: configuration, presenter: presenter)
+    .run(callback: callback)
 }
 
-public func authenticate(withSessionToken sessionToken: String) -> Authenticate {
-    return Authenticate(sessionToken: sessionToken)
+public func signOutOfOkta(from presenter: UIViewController, callback: @escaping ((OktaError?) -> Void)) {
+    SignOutTask(config: configuration, presenter: presenter)
+    .run { _, error in callback(error) }
 }
 
-public func isAuthenticated() -> Bool {
-    if tokens?.accessToken != nil {
-        return true
-    }
-    return false
+public func authenticate(withSessionToken sessionToken: String, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
+    AuthenticateTask(config: configuration, sessionToken: sessionToken)
+    .run(callback: callback)
 }
 
 public func clear() {
     // Clear auth state
     tokens?.clear()
 }
 
-public func introspect() -> Introspect {
+public func introspect(token: String?, callback: @escaping (Bool?, OktaError?) -> Void) {
```

Comment:
Any plans to move token's related functionality to TokenManager?

Reply:
Yes. This is my next task. Didn't want to enlarge this PR.